BlueTracker.SDK.Performance/Clients/ShipClient.cs

```diff
@@ -0,0 +1,100 @@
+
+namespace BlueTracker.SDK.Performance.Clients
+{
+    /// <inheritdoc />
+    /// <summary>
+    /// Client for getting, creating and updating ship information.
+    /// </summary>
+    public class ShipClient : ApiWrapper
+    {
+        /// <summary>
+        /// Creates a new ShipClient instance.
+        /// </summary>
+        /// <param name="authorization">The API token.</param>
+        /// <remarks>
+        /// The key BlueCloud_ApiKey is used to specify the API token.
+        /// </remarks>
+        public ShipClient(string authorization) :
+            base(authorization)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new ShipClient instance.
+        /// </summary>
+        /// <param name="serverAddress">The server address.</param>
+        /// <param name="authorization">The API token.</param>
+        /// <remarks>
+        /// The key BlueCloud_ApiKey is used to specify the API token, the key BlueCloud_ServerAddress is used to set the
+        /// service address. If the service address is not specified as constructor parameter,
+        /// the default service address will be used.
+        /// </remarks>
+        public ShipClient(string serverAddress, string authorization) :
+            base(serverAddress, authorization)
+        {
+        }
+
+        /// <summary>
+        /// Get a specific ship object.
+        /// </summary>
+        /// <param name="imoNumber">IMO number of the ship.</param>
+        /// <returns>The ship object.</returns>
+        public Ship GetSpecific(int imoNumber)
+        {
+            var route = $"/api/v1/ships/{imoNumber}";
+            var ret = GetObject<Ship>(route);
+            return ret;
+        }
+
+        /// <summary>
+        /// Returns a pages list of ships.
+        /// </summary>
+        /// <returns>
+        /// A paged list of ship object, which belong to the company associated with the
+        /// used API key.
+        /// </returns>
+        public PagedSearchResult<ShipShort> GetAll(int page = 0, int pageSize = 20)
+        {
+            var route = $"/api/v1/ships?page={page}&pageSize={pageSize}";
+            var ret = GetObject<PagedSearchResult<ShipShort>>(route);
+            return ret;
+        }
+
+        /// <summary>
+        /// Enables or disables the batch mode of the ships data interface.
+        /// </summary>
+        /// <param name="imoNumber">IMO number of the vessel.</param>
+        /// <param name="enableBatchMode">True to enable batch mode; false to disable batch mode.</param>
+        /// <returns>The ship object.</returns>
+        /// <remarks>
+        /// When batch mode is enabled, data items pushed to the vessel do not trigger an immediate re-calculation.
+        /// This is useful, if big amounts of items needs to be pushed. When the batch mode is disabled, a full
+        /// re-calculation is triggered.
+        /// </remarks>
+        public Ship SetBatchMode(int imoNumber, bool enableBatchMode)
```

Comment:
This endpoint is also in the ReportClient. Please decide whether we need it in both clients.

Reply:
Done.